sensor.py

```diff
@@ -106,3 +123,95 @@ def icon(self):
     @property
     def device_class(self):
         return DEVICE_CLASS_BATTERY
+
+
+class HSTemperature(HSSensor):
+    """Representation of a HomeSeer temperature sensor device."""
+
+    @property
+    def unit_of_measurement(self):
+        if self._units is not None:
+            if str(self._units) == 'imperial':
+                return "°F"
+            elif str(self._units) == 'metric':
+                return "°C"
+            else:
+                return "?"
+        else:
+            return '?'
+
+    @property
+    def icon(self):
+        if self._units is not None:
+            if str(self._units) == 'imperial':
+                if self.state < 50:
+                    return 'mdi:thermomter-low'
+                elif self.state > 90:
+                    return 'mdi:thermometer-high'
+                else:
+                    return 'mdi:thermometer'
+            elif str(self._units) == 'metric':
+                if self.state < 10:
+                    return 'mdi:thermomter-low'
+                elif self.state > 33:
+                    return 'mdi:thermometer-high'
+                else:
+                    return 'mdi:thermometer'
+            else:
+                return 'mdi:thermometer'
+        else:
+            return 'mdi: thermometer'
+
+    @property
+    def device_class(self):
+        return DEVICE_CLASS_TEMPERATURE
+
+
+class HSHumidity(HSSensor):
+    """Representation of a HomeSeer humidity sensor device."""
+
+    @property
+    def unit_of_measurement(self):
+        return '%'
+
+    @property
+    def device_class(self):
+        return DEVICE_CLASS_HUMIDITY
+
+
+class HSLuminance(HSSensor):
+    """Representation of a HomeSeer light level sensor device."""
+
+    @property
+    def unit_of_measurement(self):
+        return '%'
+
+    @property
+    def device_class(self):
+        return DEVICE_CLASS_ILLUMINANCE
+
+
+class HSFanState(HSSensor):
+    """Representation of a HomeSeer HVAC fan state sensor device."""
+
+    @property
+    def icon(self):
+        if self.state == 0:
+            return 'mdi:fan-off'
+        else:
+            return 'mdi:fan'
```

Comment:
If this is simply on/off sensor, it should be a binary sensor rather than a sensor.

Reply:
No. It can be a value in the range of 0 - 8, with 0 being 'off' and 1-8 being some state of 'on'.  I elected not to cover all 'on' states just for an icon display.